lib/anyway/loaders/doppler.rb

```diff
@@ -0,0 +1,47 @@
+# frozen_string_literal: true
+
+require 'uri'
+require 'net/http'
+require "json"
+
+module Anyway
+  using RubyNext
+
+  module Loaders
+    class Doppler < Base
+      DOPPLER_REQUEST_ERROR = Class.new(StandardError)
+      DOPPLER_JSON_FORMAT_URL = "https://api.doppler.com/v3/configs/config/secrets/download"
+
+      def call(**_options)
+        trace!(:doppler) do
+          parse_doppler_response
+        end
+      end
+
+      private
+
+      def parse_doppler_response
+        response = fetch_doppler_config
+
+        unless response.is_a?(Net::HTTPSuccess)
+          raise DOPPLER_REQUEST_ERROR, "#{response.code} #{response.message}"
+        end
+
+        JSON.parse(response.read_body)
```

Comment:
As far as I understand, the result would contain and ENV-like hash. From the documentation example:

```json
{
  "STRIPE": "sk_test_9YxLnoLDdvOPn2dfjBVPB",
  "ALGOLIA": "N9TOPUCTO",
  "DATABASE": "brian@aws.dynamodb.com:9876",
  "USER": "brian"
}
```

We should treat it as ENV, too. That is, if we want to configure Stripe:

```ruby
# Doppler data
{
  "STRIPE_TOKEN": "sk_test_9YxLnoLDdvOPn2dfjBVPB",
  "ALGOLIA_PROJECT_ID": "N9TOPUCTO",
}

StripeConfig.new.token #=> sk_test_9YxLnoLDdvOPn2dfjBVPB
AlgoliaConfig.new.token #=> N9TOPUCTO
```

Similarly, we should take care of type casting here.

We already have the required functionality here: https://github.com/palkan/anyway_config/blob/master/lib/anyway/env.rb

What if we could make the Env class more generic and support any ENV-like data container, not only `ENV`?

Reply:
thank you!

if I understand correct, it`s done

Now Anyway:Env can get the argument `env_container` with default value ENV. 
but maybe be better another name for the argument, for example `container` or `payload_container`

```
Anyway::Env.new(type_cast: ::Anyway::NoCast, env_container: env_payload)
```

And, Doppler loader repeat this (as in Anyway::Loader::Env), but with setting env_container

```
env = ::Anyway::Env.new(type_cast: ::Anyway::NoCast, env_container: env_payload)

env.fetch_with_trace(env_prefix).then do |(conf, trace)|
  Tracing.current_trace&.merge!(trace)
  conf
end
```

Of course, we can prevent this little duplicate code, but I think, that this is not the case, when we should do this (but I can wrong)